apps/frontend/src/pages/people/[username].tsx

```diff
@@ -361,169 +376,37 @@ const UserProfilePage: React.FC = () => {
     setRefreshLoading(true);
-    if (!connection || !user) {
+    if (!connection || !user || !session) {
       setRefreshLoading(false);
       return;
     }
 
     try {
-      let tensions: string[] = [];
-      if (user.userData.tensionsRating?.revealAnswers) {
-        const tensionData = tensionPairs.map((tension, index) =>
-          user.userData.tensionsRating!.tensionRating[index] < 50
-            ? tension[0]
-            : tension[1]
-        );
-        tensions = await hashCommit(
-          user.encryptionPrivateKey,
-          connection.user.encryptionPublicKey,
-          tensionData
-        );
-      }
-
-      const contactData = Object.keys(user.connections);
-      const contacts = await hashCommit(
-        user.encryptionPrivateKey,
-        connection.user.encryptionPublicKey,
-        contactData
-      );
-
-      // Devcon events
-      let devconEventTitles: string[] = [];
-      let devconEvents: string[] = [];
-      if (user.userData?.devcon?.schedule) {
-        devconEventTitles = user.userData.devcon.schedule.map(event => event.title);
-        devconEvents = await hashCommit(
-          user.encryptionPrivateKey,
-          connection.user.encryptionPublicKey,
-          devconEventTitles
-        );
-      }
-
-      // HERE: any ability to order on frequency?
-      let programmingLangs: string[] = [];
-      if (user.userData?.github?.programmingLanguages?.value) {
-        const languages = Object.keys(user.userData?.github?.programmingLanguages?.value);
-        programmingLangs = await hashCommit(
-          user.encryptionPrivateKey,
-          connection.user.encryptionPublicKey,
-          languages
-        );
-      }
-
-      let starredRepos: string[] = [];
-      if (user.userData?.github?.starredRepos?.value) {
-        starredRepos = await hashCommit(
-          user.encryptionPrivateKey,
-          connection.user.encryptionPublicKey,
-          user.userData?.github?.starredRepos?.value
-        );
-      }
-
-      const [secretHash] = await hashCommit(
-        user.encryptionPrivateKey,
-        connection.user.encryptionPublicKey,
-        [""]
-      );
-
-      const response = await fetch(
-        `${BASE_API_URL}/user/refresh_intersection`,
-        {
-          method: "POST",
-          headers: {
-            "Content-Type": "application/json",
-          },
-          body: JSON.stringify({
-            secretHash,
-            index: user.userData.username < connection.user.username ? 0 : 1,
-            intersectionState: {
-              tensions,
-              contacts,
-              devconEvents,
-              programmingLangs,
-              starredRepos,
-            },
-          }),
-        }
-      );
-
-      if (!response.ok) {
-        const errorData = await response.json();
-        throw new Error(`HTTP error! status: ${errorData.error}`);
-      }
+      const newVerifiedIntersection = await refreshPSI(user, connection);
```

Comment:
Deleted code moved into helper method. Helper method is used both by clicking Discover / Refresh button and by automatic PSI triggered by socket push notification.